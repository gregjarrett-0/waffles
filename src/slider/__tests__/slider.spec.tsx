import { useState } from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Slider } from '../index';

function TestSliderWithSingleInput() {
  const [value, setValue] = useState([30]);

  return (
    <Slider
      showInputs
      value={value}
      onChange={(value) => setValue(value)}
      aria-label="Slider with single input"
    />
  );
}

function TestSliderWithInputs() {
  const [value, setValue] = useState([30, 60]);

  return (
    <Slider
      showInputs
      value={value}
      onChange={(value) => setValue(value)}
      aria-label="Slider with inputs"
    />
  );
}

describe('Slider', () => {
  it('renders with one handle, all labels, and accessible aria attributes', () => {
    const { getByTestId, getByRole, getByText } = render(
      <Slider value={[50]} onChange={jest.fn()} aria-label="Test slider" />,
    );

    const track = getByTestId('slider-track');
    const handle = getByRole('slider');
    const currentValueLabel = getByText('50');
    const minLabel = getByText('0');
    const maxLabel = getByText('100');

    expect(track).toBeInTheDocument();
    expect(handle).toBeInTheDocument();
    expect(handle).toHaveAttribute('aria-valuenow', '50');
    expect(handle).toHaveAttribute('aria-valuemin', '0');
    expect(handle).toHaveAttribute('aria-valuemax', '100');
    expect(handle).toHaveAttribute('aria-label', 'Test slider');
    expect(currentValueLabel).toBeInTheDocument();
    expect(minLabel).toBeInTheDocument();
    expect(maxLabel).toBeInTheDocument();
  });

  it('renders with two handles, and all labels', () => {
    const { getByTestId, getAllByRole, getByText } = render(
      <Slider value={[20, 50]} onChange={jest.fn()} aria-label="Test slider" />,
    );

    const track = getByTestId('slider-track');
    const handles = getAllByRole('slider');
    const firstValueLabel = getByText('20');
    const secondValueLabel = getByText('50');
    const minLabel = getByText('0');
    const maxLabel = getByText('100');

    expect(track).toBeInTheDocument();
    expect(handles).toHaveLength(2);
    expect(handles[0]).toHaveAttribute('aria-valuenow', '20');
    expect(handles[1]).toHaveAttribute('aria-valuenow', '50');
    expect(firstValueLabel).toBeInTheDocument();
    expect(secondValueLabel).toBeInTheDocument();
    expect(minLabel).toBeInTheDocument();
    expect(maxLabel).toBeInTheDocument();
  });

  it('renders with custom min and max', () => {
    const { getByText } = render(
      <Slider
        min={-100}
        max={200}
        value={[50]}
        onChange={jest.fn()}
        aria-label="Test slider"
      />,
    );

    const minLabel = getByText('-100');
    const maxLabel = getByText('200');

    expect(minLabel).toBeInTheDocument();
    expect(maxLabel).toBeInTheDocument();
  });

  it('renders with customized labels', () => {
    const { getByText, getByRole } = render(
      <Slider
        value={[50]}
        onChange={jest.fn()}
        aria-label="Test slider"
        formatLabel={(value) => {
          return `${value} $`;
        }}
      />,
    );

    const minLabel = getByText('0 $');
    const maxLabel = getByText('100 $');
    const valueLabel = getByText('50 $');
    const handle = getByRole('slider');

    expect(minLabel).toBeInTheDocument();
    expect(maxLabel).toBeInTheDocument();
    expect(valueLabel).toBeInTheDocument();
    expect(handle).toHaveAttribute('aria-valuetext', '50 $');
  });

  describe('when optional inputs are displayed', () => {
    it('renders one input when one value is provided', () => {
      const { container } = render(
        <Slider
          showInputs
          value={[50]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const input = container.querySelector('input');

      expect(input).toBeInTheDocument();
      expect(input).toHaveAttribute('value', '50');
      expect(input).toHaveAttribute('aria-label', 'Test slider');
    });

    it('renders two inputs when two values are provided', () => {
      const { container } = render(
        <Slider
          showInputs
          value={[30, 60]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const inputs = container.querySelectorAll('input');

      expect(inputs).toHaveLength(2);
      expect(inputs[0]).toHaveAttribute('value', '30');
      expect(inputs[1]).toHaveAttribute('value', '60');
    });

    it('when slider is disabled inputs are disabled too', () => {
      const { container } = render(
        <Slider
          showInputs
          disabled
          value={[30, 60]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const inputs = container.querySelectorAll('input');

      expect(inputs[0]).toBeDisabled();
      expect(inputs[1]).toBeDisabled();
    });

    it("when input's value has changed, slider's value is updated too", () => {
      const { container, getByRole } = render(<TestSliderWithSingleInput />);

      const input = container.querySelector('input');
      input && fireEvent.change(input, { target: { value: '42' } });

      const handle = getByRole('slider');

      expect(input).toBeInTheDocument();
      expect(handle).toHaveAttribute('aria-valuenow', '42');
    });

    it("for single input when it's value is less than slider min value, show error", () => {
      const { container } = render(<TestSliderWithSingleInput />);

      const input = container.querySelector('input');
      input && fireEvent.change(input, { target: { value: '-10' } });

      expect(input).toHaveAttribute('aria-invalid', 'true');
    });

    it("for single input when it's value is greater than slider max value, show error", () => {
      const { container } = render(<TestSliderWithSingleInput />);

      const input = container.querySelector('input');
      input && fireEvent.change(input, { target: { value: '200' } });

      expect(input).toHaveAttribute('aria-invalid', 'true');
    });

    it("for single input when it's value doesn't overlap with step, show error", () => {
      const { container } = render(<TestSliderWithSingleInput />);

      const input = container.querySelector('input');
      input && fireEvent.change(input, { target: { value: '10.5' } });

      expect(input).toHaveAttribute('aria-invalid', 'true');
    });

    it("for two inputs when first one's value is less than slider min value, show error", () => {
      const { container } = render(<TestSliderWithInputs />);

      const inputs = container.querySelectorAll('input');
      inputs[0] && fireEvent.change(inputs[0], { target: { value: '-10' } });

      expect(inputs[0]).toHaveAttribute('aria-invalid', 'true');
    });

    it("for two inputs when first one's value is greater than second one, show error", () => {
      const { container } = render(<TestSliderWithInputs />);

      const inputs = container.querySelectorAll('input');
      inputs[0] && fireEvent.change(inputs[0], { target: { value: '70' } });

      expect(inputs[0]).toHaveAttribute('aria-invalid', 'true');
    });

    it("for two inputs when second one's value is less than first one, show error", () => {
      const { container } = render(<TestSliderWithInputs />);

      const inputs = container.querySelectorAll('input');
      inputs[1] && fireEvent.change(inputs[1], { target: { value: '10' } });

      expect(inputs[1]).toHaveAttribute('aria-invalid', 'true');
    });

    it("for two inputs when second one's value is greater than slider max value, show error", () => {
      const { container } = render(<TestSliderWithInputs />);

      const inputs = container.querySelectorAll('input');
      inputs[1] && fireEvent.change(inputs[1], { target: { value: '200' } });

      expect(inputs[1]).toHaveAttribute('aria-invalid', 'true');
    });
  });

  describe('renders snapshot', () => {
    it('with one handle', () => {
      const { container } = render(
        <Slider
          inverted
          value={[30]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });

    it('with two handles', () => {
      const { container } = render(
        <Slider
          inverted
          value={[20, 40]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });

    it('with all labels hidden', () => {
      const { container } = render(
        <Slider
          hideLabels
          value={[20, 40]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });

    it('with inputs instead of labels', () => {
      const { container } = render(
        <Slider
          showInputs
          value={[20, 40]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });

    it('of inverted', () => {
      const { container } = render(
        <Slider
          inverted
          value={[50]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });

    it('of disabled', () => {
      const { container } = render(
        <Slider
          disabled
          value={[50]}
          onChange={jest.fn()}
          aria-label="Test slider"
        />,
      );

      const slider = container.firstChild;
      expect(slider).toMatchSnapshot();
    });
  });
});
