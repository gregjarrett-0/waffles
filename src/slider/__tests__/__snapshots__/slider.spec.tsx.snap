// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders snapshot of disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-size: 20px;
  font-weight: 800;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 0.6;
}

.emotion-3 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-5 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #009BD8;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: -4px;
}

.emotion-7 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #65707C;
  font-size: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      50
    </span>
  </div>
  <div
    class="emotion-2"
    data-testid="slider-track"
    style="transform: scale(1); cursor: inherit;"
  >
    <div
      class="emotion-3"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      aria-valuetext="50"
      class="emotion-4"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: inherit; user-select: none; transform: translate(NaNpx, NaNpx);"
    >
      <div
        class="emotion-5"
      />
    </div>
  </div>
  <div
    class="emotion-6"
  >
    <span
      class="emotion-7"
    >
      0
    </span>
    <span
      class="emotion-7"
    >
      100
    </span>
  </div>
</div>
`;

exports[`Slider renders snapshot of inverted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 800;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 1;
}

.emotion-3 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-4:hover>div {
  box-shadow: 0 0 0 3px #06BDFC;
}

.emotion-5 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #06BDFC;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: -4px;
}

.emotion-7 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #9BA3AB;
  font-size: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      50
    </span>
  </div>
  <div
    class="emotion-2"
    data-testid="slider-track"
    style="transform: scale(1); cursor: pointer;"
  >
    <div
      class="emotion-3"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      aria-valuetext="50"
      class="emotion-4"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-5"
      />
    </div>
  </div>
  <div
    class="emotion-6"
  >
    <span
      class="emotion-7"
    >
      0
    </span>
    <span
      class="emotion-7"
    >
      100
    </span>
  </div>
</div>
`;

exports[`Slider renders snapshot with all labels hidden 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 1;
}

.emotion-1 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-2:hover>div {
  box-shadow: 0 0 0 3px #009BD8;
}

.emotion-3 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #009BD8;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

<div>
  <div
    class="emotion-0"
    data-testid="slider-track"
    style="transform: scale(1); cursor: inherit;"
  >
    <div
      class="emotion-1"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="40"
      aria-valuemin="0"
      aria-valuenow="20"
      aria-valuetext="20"
      class="emotion-2"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-3"
      />
    </div>
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="20"
      aria-valuenow="40"
      aria-valuetext="40"
      class="emotion-2"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 1; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-3"
      />
    </div>
  </div>
</div>
`;

exports[`Slider renders snapshot with inputs instead of labels 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 60px 60px;
  gap: calc(100% - 120px);
}

.emotion-1 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-1::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  font-size: 14px;
  color: #05192D;
  background-color: #FFFFFF;
  padding-left: 12px;
  padding-right: 12px;
}

.emotion-2:disabled {
  cursor: default;
}

.emotion-2:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-2::-webkit-search-decoration,
.emotion-2::-webkit-search-cancel-button,
.emotion-2::-webkit-search-results-button,
.emotion-2::-webkit-search-results-decoration {
  display: none;
}

.emotion-2::-ms-clear,
.emotion-2::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-2::-webkit-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-2::-moz-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-2:-ms-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-2::placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 1;
}

.emotion-6 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-7:hover>div {
  box-shadow: 0 0 0 3px #009BD8;
}

.emotion-8 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #009BD8;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: -4px;
}

.emotion-12 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #65707C;
  font-size: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <input
        aria-invalid="false"
        aria-label="Test slider"
        class="emotion-2"
        value="20"
      />
    </div>
    <div
      class="emotion-1"
    >
      <input
        aria-invalid="false"
        aria-label="Test slider"
        class="emotion-2"
        value="40"
      />
    </div>
  </div>
  <div
    class="emotion-5"
    data-testid="slider-track"
    style="transform: scale(1); cursor: inherit;"
  >
    <div
      class="emotion-6"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="40"
      aria-valuemin="0"
      aria-valuenow="20"
      aria-valuetext="20"
      class="emotion-7"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-8"
      />
    </div>
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="20"
      aria-valuenow="40"
      aria-valuetext="40"
      class="emotion-7"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 1; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-8"
      />
    </div>
  </div>
  <div
    class="emotion-11"
  >
    <span
      class="emotion-12"
    >
      0
    </span>
    <span
      class="emotion-12"
    >
      100
    </span>
  </div>
</div>
`;

exports[`Slider renders snapshot with one thumb 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 800;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 1;
}

.emotion-3 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-4:hover>div {
  box-shadow: 0 0 0 3px #06BDFC;
}

.emotion-5 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #06BDFC;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: -4px;
}

.emotion-7 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #9BA3AB;
  font-size: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      30
    </span>
  </div>
  <div
    class="emotion-2"
    data-testid="slider-track"
    style="transform: scale(1); cursor: pointer;"
  >
    <div
      class="emotion-3"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="30"
      aria-valuetext="30"
      class="emotion-4"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-5"
      />
    </div>
  </div>
  <div
    class="emotion-6"
  >
    <span
      class="emotion-7"
    >
      0
    </span>
    <span
      class="emotion-7"
    >
      100
    </span>
  </div>
</div>
`;

exports[`Slider renders snapshot with two thumbs 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 800;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: calc(100% - 12px);
  margin-left: 6px;
  height: 28px;
  opacity: 1;
}

.emotion-4 {
  width: calc(100% + 12px);
  margin-left: -6px;
  margin-right: -6px;
  height: 4px;
  border-radius: 2px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 28px;
  height: 28px;
  outline: 0;
}

.emotion-5:hover>div {
  box-shadow: 0 0 0 3px #06BDFC;
}

.emotion-6 {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background-color: #06BDFC;
  -webkit-transition: box-shadow 200ms ease-out;
  transition: box-shadow 200ms ease-out;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: -4px;
}

.emotion-10 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #9BA3AB;
  font-size: 12px;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      20
    </span>
    <span
      class="emotion-1"
    >
      40
    </span>
  </div>
  <div
    class="emotion-3"
    data-testid="slider-track"
    style="transform: scale(1); cursor: inherit;"
  >
    <div
      class="emotion-4"
    />
    <div
      aria-label="Test slider"
      aria-valuemax="40"
      aria-valuemin="0"
      aria-valuenow="20"
      aria-valuetext="20"
      class="emotion-5"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-6"
      />
    </div>
    <div
      aria-label="Test slider"
      aria-valuemax="100"
      aria-valuemin="20"
      aria-valuenow="40"
      aria-valuetext="40"
      class="emotion-5"
      draggable="false"
      role="slider"
      style="position: absolute; z-index: 1; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);"
      tabindex="0"
    >
      <div
        class="emotion-6"
      />
    </div>
  </div>
  <div
    class="emotion-9"
  >
    <span
      class="emotion-10"
    >
      0
    </span>
    <span
      class="emotion-10"
    >
      100
    </span>
  </div>
</div>
`;
